Dockerfile

```diff
@@ -0,0 +1,40 @@
+FROM rust:1.69-bullseye AS chef
+RUN rustup update stable --no-self-update \
+    && rustup default stable \
+    && rustup target add wasm32-wasi \
+    && rustup target add wasm32-unknown-unknown
+RUN mkdir -p /usr/local/protobuf \
+    && wget https://github.com/protocolbuffers/protobuf/releases/download/v22.3/protoc-22.3-linux-x86_64.zip \
+    && unzip protoc-22.3-linux-x86_64.zip -d /usr/local/protobuf \
+    && rm protoc-22.3-linux-x86_64.zip
+ENV PATH="${PATH}:/usr/local/protobuf/bin"
+RUN cargo install cargo-chef --locked
+
+FROM chef AS planner
+WORKDIR /usr/src/bytecodealliance/registry
+COPY . .
+RUN cargo chef prepare --recipe-path recipe.json
+
+FROM chef AS builder
+WORKDIR /usr/src/bytecodealliance/registry
+COPY --from=planner /usr/src/bytecodealliance/registry/recipe.json ./
+RUN cargo chef cook --release --workspace --recipe-path recipe.json
+COPY . .
+RUN cargo build --release --workspace
+
+FROM gcr.io/distroless/cc AS warg-server
```

Comment:
It is probably seen more often to have a separate Dockerfile for a separate binary, but having separate targets here are maybe helping speed and simplify integration tests?

Reply:
@peterhuene has done the most integration testing work, so I'll let him speak to this.